src/test/groovy/ee/tuleva/onboarding/mandate/batch/MandateBatchServiceTest.java

```diff
@@ -0,0 +1,136 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class MandateBatchServiceTest {
+
+  @Mock private MandateBatchRepository mandateBatchRepository;
+
+  @Mock private MandateFileService mandateFileService;
+  @Mock private GenericMandateService genericMandateService;
+
+  @InjectMocks private MandateBatchService mandateBatchService;
+
+  @Test
+  @DisplayName("return MandateBatch by id and user when all mandates belong to the user")
+  void returnMandateBatch() {
+    var user = sampleUser().build();
+    var mandate1 = sampleFundPensionOpeningMandate();
+    var mandate2 = samplePartialWithdrawalMandate();
+
+    var mandateBatch =
+        MandateBatchFixture.aMandateBatch().mandates(List.of(mandate1, mandate2)).build();
+
+    when(mandateBatchRepository.findById(any())).thenReturn(Optional.of(mandateBatch));
+
+    Optional<MandateBatch> result = mandateBatchService.getByIdAndUser(1L, user);
+
+    assertThat(result.isPresent()).isTrue();
+  }
+
+  @Test
+  @DisplayName("return empty when all mandates do not match the user")
+  void mandatesDontMatch() {
+    var user = sampleUser().build();
+    var differentUser = sampleUser().id(2L).build();
+
+    var mandate1 = sampleFundPensionOpeningMandate();
+    var mandate2 = samplePartialWithdrawalMandate();
+    mandate2.setUser(differentUser);
+
+    var mandateBatch =
+        MandateBatchFixture.aMandateBatch().mandates(List.of(mandate1, mandate2)).build();
+
+    when(mandateBatchRepository.findById(any())).thenReturn(Optional.of(mandateBatch));
+
+    Optional<MandateBatch> result = mandateBatchService.getByIdAndUser(1L, user);
+
+    assertTrue(result.isEmpty());
+  }
+
+  @Test
+  @DisplayName("return mandate batch content files for user")
+  void getMandateBatchContentFiles() {
+    var mandate1 = sampleFundPensionOpeningMandate();
+    var mandate2 = samplePartialWithdrawalMandate();
+    var user = mandate1.getUser();
+
+    var mandateBatch =
+        MandateBatchFixture.aMandateBatch().mandates(List.of(mandate1, mandate2)).build();
+
+    when(mandateBatchRepository.findById(1L)).thenReturn(Optional.of(mandateBatch));
+
+    when(mandateFileService.getMandateFiles(mandate1))
+        .thenReturn(List.of(new SignatureFile("file.html", "text/html", new byte[0])));
+    when(mandateFileService.getMandateFiles(mandate2))
+        .thenReturn(List.of(new SignatureFile("file2.html", "text/html", new byte[0])));
+
+    List<SignatureFile> result = mandateBatchService.getMandateBatchContentFiles(1L, user);
+
+    assertThat(result.size()).isEqualTo(2);
+    verify(mandateBatchRepository, times(1)).findById(1L);
+    verify(mandateFileService, times(1)).getMandateFiles(mandate1);
+    verify(mandateFileService, times(1)).getMandateFiles(mandate2);
```

Comment:
These kinds of mock interaction verifications are very fragile (they break as soon as you refactor the code a bit - i.e. introduce new collaborator classes or merge some collaborators into 1). 

Fragile tests stop us from refactoring code conveniently.

The best kind of assertions are purely functional (i.e. they only assert the output of the function).